spinta/cli/sync.py

```diff
@@ -1,161 +1,70 @@
-        )
+        dataset_id = extract_dataset_id(response_get_dataset, "list")
+        update_dsa(base_path, headers, dataset_id)
     elif response_get_dataset.status_code == HTTPStatus.NOT_FOUND:
-        response_post_dataset = requests.post(
-            f"{base_path}/Dataset/",
-            headers=authorization_headers,
-            data={
-                "title": dataset_name,
-                "description": "",
-                "name": dataset_name,
-            }
-        )
-
-        if response_post_dataset.status_code != HTTPStatus.CREATED:
-            raise UnexpectedAPIResponse(
-                operation="Create dataset",
-                expected_status_code=HTTPStatus.CREATED,
-                response_status_code=response_post_dataset.status_code,
-                response_data=_format_error_response_data(response_post_dataset.json()),
-            )
-
-        dataset_id = response_post_dataset.json().get(IDENTIFIER)
-        if not dataset_id:
-            raise UnexpectedAPIResponseData(
-                operation=f"Retrieve dataset `{IDENTIFIER}`",
-                context=f"Dataset did not return the `{IDENTIFIER}` field which can be used to identify the dataset."
-            )
-        response_post_distribution = requests.post(
-            f"{base_path}/Distribution/",
-            headers=authorization_headers,
-            data={
-                "dataset": dataset_id,
-                "title": dataset_name,
-            },
-            files={
-                "file": (manifests[0].path, BytesIO(file_bytes), CONTENT_TYPE_TEXT_CSV),
-            }
-        )
-
-        if response_post_distribution.status_code != HTTPStatus.CREATED:
-            raise UnexpectedAPIResponse(
-                operation="Create dataset distribution",
-                expected_status_code=HTTPStatus.CREATED,
-                response_status_code=response_post_distribution.status_code,
-                response_data=_format_error_response_data(response_post_distribution.json()),
-            )
-
-        response_create_dsa = requests.post(
-            f"{base_path}/Dataset/{dataset_id}/dsa/",
-            headers={"Content-Type": CONTENT_TYPE_TEXT_CSV, **authorization_headers},
-            data=content,
-        )
-
-        if response_create_dsa.status_code != HTTPStatus.NO_CONTENT:
-            raise UnexpectedAPIResponse(
-                operation="Create DSA for dataset",
-                expected_status_code=HTTPStatus.CREATED,
-                response_status_code=response_create_dsa.status_code,
-                response_data=_format_error_response_data(response_create_dsa.json()),
-            )
+        response_create_dataset = create_dataset(base_path, headers, dataset_name)
+        dataset_id = extract_dataset_id(response_create_dataset, "detail")
+        create_distribution(base_path, headers, dataset_name, file_bytes, dataset_id, manifest_objects)
+        create_dsa(base_path, headers, dataset_id, content)
```

Comment:
transaction (atomic) would be nice

Reply:
Could you explain what would be the point of that?

The calls & database changes are to an external API, we can not guarantee atomicity for the other system without additional changes, or am I wrong?